### ISSUE TITLE
Can't add signature to an account on Mac OS

### ISSUE THREAD

**Issue by bigmmind** [nylas/nylas-mail#3286]
See the video, choosing account from the default list does nothing. Apart from that clicking plus creates two new signatures.

https://drive.google.com/open?id=0B_S4E859g0NoQUFPeUtEOHBPMnc

Here is my system info

![screen shot 2017-02-15 at 12 14 07](https://cloud.githubusercontent.com/assets/23629885/22967829/1385918a-f379-11e6-8873-552f4a55191c.png)

![screen shot 2017-02-15 at 12 20 29](https://cloud.githubusercontent.com/assets/23629885/22967848/2ccf9b90-f379-11e6-840c-4b0ec3332747.png)

![screen shot 2017-02-15 at 12 14 18](https://cloud.githubusercontent.com/assets/23629885/22967830/155a0392-f379-11e6-9d1c-b615e3a0bcb3.png)

**Comment by GMory** [comment]
Experiencing the same here.

**Comment by joshrps** [comment]
Also experiencing this. Screen recording: http://d.pr/i/KnMw

<img width="251" alt="screen shot 2017-02-15 at 11 31 49 am" src="https://cloud.githubusercontent.com/assets/4356581/22984120/696d23ee-f372-11e6-9ca3-a7175139ad4c.png">

<img width="396" alt="screen shot 2017-02-15 at 11 32 45 am" src="https://cloud.githubusercontent.com/assets/4356581/22984153/7c3444f8-f372-11e6-957b-dd97b920ef3d.png">

**Comment by derekatkinson** [comment]
Experiencing the same thing, for both inability to associate a signature with an account and new signature button creating two signatures.

![image](https://cloud.githubusercontent.com/assets/6711494/22988944/e0652f14-f381-11e6-8402-fd568a847ad9.png)

![image](https://cloud.githubusercontent.com/assets/6711494/22988920/ccc4686c-f381-11e6-861c-10f3fdc2ed63.png)

**Comment by mdelange** [comment]
Same here.
Using Nylas Mail 1.0.27 on Mac Pro 1,1 running OS X 10.9.5 (yes it's still running fine ;))

**Comment by jeromecovington** [comment]
Having this issue on Nylas Mail 1.0.27 and OS X 10.10.5.

**Comment by mallendeo** [comment]
I think the problem is this:

`src/flux/stores/signature-store.js:132`
```js
_onToggleAccount = (email) => {
  if (this.defaultSignatures[email] === this.selectedSignatureId) {
    this.defaultSignatures[email] = null
  } else {
    this.defaultSignatures[email] = this.selectedSignatureId
  }

  this.trigger()
  this._saveDefaultSignatures() // it should be this._saveSignatures() I guess?
}
```

I'm trying to test it, but I can't get sync-engine to build.

**Comment by shenanigans12** [comment]
Ditto.  Theres absolutely no way to disable the default "Nylas" signature right now

Nylas Mail 1.0.27, OS X 10.12.3

**Comment by madurimanoj** [comment]
same problem

**Comment by kelvinczech** [comment]
Same issue with Nylas Mail 1.0.27, macOS 10.12.3.

**Comment by agiantwhale** [comment]
Experiencing same issue here

**Comment by cvbourne** [comment]
Also experiencing this issue.

**Comment by kaspar-allenbach** [comment]
Jup mee too:

macOS 10.12.3 + Nylas 1.0.27

**Comment by bricehua** [comment]
Same with Nylas Version 1.0.27 (1.0.27) and Sierra 10.12.3

**Comment by bemky** [comment]
Can we get an update?

**Comment by stolinski** [comment]
Same thing here. Frustrated that I'm paying for this right now.

**Comment by edarioq** [comment]
Same, aside from the fact that I can't send emails from my custom IMAP account, Nylas just seems very buggy to me. They should label this a beta. Good thing I'm not paying anything just yet.

**Comment by sebastiang1** [comment]
Same here. Nylas Version 1.0.27 (1.0.27) + macOS Sierra 10.12.3.

**Comment by augustoms** [comment]
Same here.
Nylas Version 1.0.27 (1.0.27) + OSX El Capitan 10.11.6.

**Comment by jstejada** [comment]
Hi all! Sorry for the late response. We're going to look into this!

**Comment by adenix** [comment]
I'm able to add a signature, however, I am unable to set default signatures.

![screen shot 2017-02-23 at 9 39 51 am](https://cloud.githubusercontent.com/assets/8764760/23263387/182ee5d6-f9ac-11e6-93df-2afae3e82793.png)
![screen shot 2017-02-23 at 9 40 05 am](https://cloud.githubusercontent.com/assets/8764760/23263388/1837d8f8-f9ac-11e6-9693-6cc69235b509.png)

**Comment by adenix** [comment]
I am able to create a signature, however, I am unable to set it as default.

![screen shot 2017-02-23 at 9 39 51 am](https://cloud.githubusercontent.com/assets/8764760/23263499/6a94a310-f9ac-11e6-976b-8e0a98d56262.png)
![macos](https://cloud.githubusercontent.com/assets/8764760/23263519/79272c86-f9ac-11e6-9fcd-e0a74ae1021d.png)

**Comment by jascolby** [comment]
Same here.

**Comment by lcoghill** [comment]
Same problem. I have tried the previous suggestions of replacing the signature with an empty line and it saves correctly, but the "ad" line still appears. I also can't select either of my accounts to set the default signature for them.

**Comment by jstejada** [comment]
Hi everyone, just wanted to let you know that a fix for this will be available in the next release. Thanks for your patience!

**Comment by jstejada** [comment]
This should be fixed in version 1.0.30 which has just been released! Let me know if you continue having issues!

**Comment by adenix** [comment]
Also fixes #3303
